fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -1709,98 +1709,93 @@ private Long getLongSqlRowSet(final SqlRowSet rs, final String column) {
+            case "m_center":
+                scopedSQL = "select o.id as officeId, g.id as groupId, null as clientId, null as savingsId, null as loanId, null as entityId from m_group g "
+                        + " join m_office o on o.id = g.office_id and " + officeHierarchyCondition + " where g.id = " + appTableId;
+            break;
+            case "m_office":
+                scopedSQL = "select o.id as officeId, null as groupId, null as clientId, null as savingsId, null as loanId, null as entityId from m_office o "
+                        + " where " + officeHierarchyCondition + " and o.id = " + appTableId;
+            break;
+            case "m_product_loan":
+            case "m_savings_product":
+            case "m_share_product":
+                scopedSQL = "select null as officeId, null as groupId, null as clientId, null as savingsId, null as loanId, p.id as entityId from "
+                        + appTable + " as p WHERE p.id = " + appTableId;
+            break;
+            default:
+                throw new PlatformDataIntegrityException("error.msg.invalid.dataScopeCriteria",
+                        "Application Table: " + appTable + " not catered for in data Scoping");
+        }
         return scopedSQL;
 
     }
 
     private void validateAppTable(final String appTable) {
-
-        if (appTable.equalsIgnoreCase("m_loan")) {
-            return;
-        }
-        if (appTable.equalsIgnoreCase("m_savings_account")) {
-            return;
-        }
-        if (appTable.equalsIgnoreCase("m_client")) {
-            return;
-        }
-        if (appTable.equalsIgnoreCase("m_group")) {
-            return;
-        }
-        if (appTable.equalsIgnoreCase("m_center")) {
-            return;
-        }
-        if (appTable.equalsIgnoreCase("m_office")) {
-            return;
-        }
-        if (appTable.equalsIgnoreCase("m_product_loan")) {
-            return;
-        }
-        if (appTable.equalsIgnoreCase("m_savings_product")) {
-            return;
-        }
-        if (appTable.equalsIgnoreCase("m_share_product")) {
-            return;
+        switch (appTable.toLowerCase()) {
+            case "m_loan":
+            case "m_savings_account":
+            case "m_savings_account_transaction":
+            case "m_client":
+            case "m_group":
+            case "m_center":
+            case "m_office":
+            case "m_product_loan":
+            case "m_savings_product":
+            case "m_share_product":
+                return;
+            default:
+                throw new PlatformDataIntegrityException("error.msg.invalid.application.table", "Invalid Application Table: " + appTable,
+                        "name", appTable);
         }
-
-        throw new PlatformDataIntegrityException("error.msg.invalid.application.table", "Invalid Application Table: " + appTable, "name",
-                appTable);
     }
 
     private String mapToActualAppTable(final String appTable) {
```

Comment:
There are so many other places regarding data-tables where you should add the new type:
org.apache.fineract.infrastructure.dataqueries.data.EntityTables